### ISSUE TITLE
Rewrite the internal music code

### ISSUE THREAD

**Issue by MrJacz** [SwitchbladeBot/switchblade#776]
I'm going to use no template for this as it isn't really a feature request per say.
I'm the developer of discord.js-lavalink and I've become aware of your bot, I would be happy to help you rewrite your music code as it is very lacking at the moment. 
Snekfetch is also deprecated and unmaintained. 
I don't mean to be a pain in the ass but the code I've gone thru for music looks very unreadable and slow. after v3 of discord.js-lavalink I would be happy to help you rewrite your music system.

**Comment by pedrofracassi** [comment]
Sure, we'd love your help! @davipatury is the one responsible for most of the music code, maybe you two can talk and see where it goes :)

**Comment by davipatury** [comment]
Yeah, I was planning to rewrite it a few months ago, but then lack of time didn't help.

**Comment by metehus** [comment]
Great! hope it goes fine

**Comment by pedrofracassi** [comment]
As for snekfetch, yeah, we've been planning to stop using it for quite some time now.

By the way, would you mind joining our [community server](http://support.switchblade.xyz/) so we can discuss stuff more easily?

**Comment by MrJacz** [comment]
@pedrofracassi I can do that, sure.

**Comment by dpaiv0** [comment]
ps ;; we can replace snekfetch with [node-fetch](https://npmjs.com/package/node-fetch), and deprecate request too

**Comment by pedrofracassi** [comment]
Let's do it then!

**Comment by dpaiv0** [comment]
I am going to try to do something that can find every file that contains `require('snekfetch')` or `require('request')`

**Comment by MrJacz** [comment]
@dpaiv0 switch to `node-fetch` it comes by default in discord.js master

**Comment by pedrofracassi** [comment]
We're not using master just yet. We prefer to stick to stable

**Comment by MrJacz** [comment]
Discord.js master is by far more stable than stable, it also has a lot of bug fixes.
